### ISSUE TITLE
The client method is deprecated as of redis-rb 4.0.0, please use the new _clientmethod instead. Support for the old method will be removed in redis-namespace 2.0.

### ISSUE THREAD

**Issue by huuth** [resque/resque#1591]
Everything is working fine, but the message "The client method is deprecated as of redis-rb 4.0.0, please use the new _clientmethod instead. Support for the old method will be removed in redis-namespace 2.0." always dispayed in terminal console.

Please explain or help me fix it. Thanks

**Comment by henrik** [comment]
@huuth You can probably work around it by downgrading the redis-rb gem to version 3. In your Gemfile:

```
gem "redis", "<4"
```

Then run `bundle update redis`.

Or wait for Resque to be updated, or make a pull request :)

It seems redis-rb has removed or renamed some methods that Resque uses: https://github.com/redis/redis-rb/blob/master/CHANGELOG.md#40

**Comment by Fjool** [comment]
This warning originates in redis-namespace:

https://github.com/resque/redis-namespace/blob/master/lib/redis/namespace.rb

**Comment by jeremy** [comment]
Fixed by https://github.com/resque/redis-namespace/pull/136

**Comment by spilth** [comment]
Any timeline for a release with this fix?

**Comment by jeremy** [comment]
No timeline, @spilth. Other than Real Soon Now 😍

**Comment by joshuabaird** [comment]
Still looking at "real soon?" :)

**Comment by jeremy** [comment]
_Real_ Soon Now 🤐

**Comment by vruizext** [comment]
I had to monkey-patch `Redis::Namespace` in my initializer as a workaround, until the fix is released

```ruby
Redis::Namespace.class_eval do
  def client
    @redis._client || @redis.client
  end
end
```

**Comment by CodingAnarchy** [comment]
Any update? Not sure why this is closed when it is still an issue a year later.

**Comment by rafaelfranca** [comment]
This is already fixed on the master branch, so this is why it is closed. Feel free to point to it

**Comment by ErikMaki** [comment]
"gem install resque" and "bundle update" don't grab the latest commits. Does resque need a version change for these commands to work?

**Comment by zarnautovi** [comment]
My version is 1.27.4 (last one that bundle is fetching) and I still have those warnings. When can we expect new verison that have this fix?

**Comment by zohaib-sarwar-confiz** [comment]
I am using Redis version 4.0.2 but  I am facing same issue.

**Comment by zohaib-sarwar-confiz** [comment]
I am using Redis version 4.0.2 and my resque version is 1.27.4 but I am facing same issue.

**Comment by zazaian** [comment]
@rafaelfranca has no new release been issued? Why hasn't this fix been integrated into a modified release yet?

**Comment by will3216** [comment]
If this gets fixed I'll have to add a logging message to determine if jobs are getting processed... I vote we keep it!
![image](https://imgs.xkcd.com/comics/workflow.png)
But seriously though; I've monkey-patched it for now, but I'd be great if we could get a release with this fix built-in! Any updates on this?

**Comment by NDuggan** [comment]
@rafaelfranca  @zazaian +1 - these warnings are so noisy! Prefer to link to a released version as opposed to the `master` branch.

**Comment by pjrebsch** [comment]
Looks like this issue was resolved by #1606 which is now included in release [2.0.0](https://github.com/resque/resque/releases/tag/v2.0.0) made available on November 6, 2018.
